### ISSUE TITLE
Changes to settings.json require restart

### ISSUE THREAD

**Issue by jdhitsolutions** [microsoft/terminal#5625]
# Environment

```none
Platform ServicePack Version      VersionString
-------- ----------- -------      -------------
 Win32NT             10.0.18363.0 Microsoft Windows NT 10.0.18363.0
```

Windows Terminal 0.11.1121.0

I rebuilt my settings.json file per instructions to use the new format.
# Steps to reproduce

From Windows Terminal, open Settings and make a change to the active profile such as changing the cursor shape or background image.

# Expected behavior

In previous versions, Windows Terminal would detect the change and update the profile.

# Actual behavior

Now, I have to restart Windows Terminal completely to get the new profile settings.  Even closing and reopening the profile isn't enough.

I tried to find where this behavior changed to see if it was a design change but couldn't find anything. I prefer the immediate updating that now seems missing.

**Comment by DHowett-MSFT** [comment]
Unfortunately, I'm not able to reproduce this..

Can you share your `settings.json`?

![hot_reload_works_file_for_me](https://user-images.githubusercontent.com/14316954/80524736-f3d7a980-8944-11ea-833a-d9cd46dfa8e8.gif)

**Comment by jdhitsolutions** [comment]
I renamed the old file and let WT create a new one. Then I pasted my old profiles back in.

```json
// This file was initially generated by Windows Terminal 0.11.1121.0
// It should still be usable in newer versions, but newer versions might have additional
// settings, help text, or changes that you will not see unless you clear this file
// and let us generate a new one for you.

// To view the default settings, hold "alt" while clicking on the "Settings" button.
// For documentation on these settings, see: https://aka.ms/terminal-documentation
{
    "$schema": "https://aka.ms/terminal-profiles-schema",

    "defaultProfile": "{993855ad-b0eb-4f3d-8370-1a8d5b53abb5}",

    // You can add more global application settings here.
    // To learn more about global settings, visit https://aka.ms/terminal-global-settings

    // If enabled, selections are automatically copied to your clipboard.
    "copyOnSelect": false,

    // If enabled, formatted data is also copied to your clipboard
    "copyFormatting": false,

    // A profile specifies a command to execute paired with information about how it should look and feel.
    // Each one of them will appear in the 'New Tab' dropdown,
    //   and can be invoked from the commandline with `wt.exe -p xxx`
    // To learn more about profiles, visit https://aka.ms/terminal-profile-settings
    "profiles":
    {
        "defaults":
        {
            // Put settings here that you want to apply to all profiles.
            "historySize": 9001,
            "padding": "0, 0, 0, 0",
            "snapOnInput": true,
            "closeOnExit": true
        },
        "list":
        [
            {
                // PowerShell 7 - ok
                "acrylicOpacity": 0.60000002384185791,
                "colorScheme": "Campbell",
                "commandline": "C:\\Program Files\\PowerShell\\7\\pwsh.exe -nologo",
                "cursorColor": "#FFFFFF",
                "cursorShape": "underscore",
                "fontFace": "Cascadia Code",
                "fontSize": 14,
                "guid": "{993855ad-b0eb-4f3d-8370-1a8d5b53abb5}",
                "icon": "ms-appx:///ProfileIcons//pwsh-preview.scale-150.png",
                "name": "PowerShell 7",
                "startingDirectory": "C:\\",
                "tabTitle": "PowerShell 7",
                "backgroundImage": "C:\\Users\\Jeff\\OneDrive\\terminalthumbs\\snover-psbomb-thumb.png",
                "backgroundImageAlignment": "bottomRight",
                "backgroundImageStretchMode": "none",
                "useAcrylic": true
            },
            {
                //PowerShell 7 No Profile -ok
                "acrylicOpacity": 0.60000002384185791,
                "closeOnExit": true,
                "colorScheme": "Campbell",
                "commandline": "C:\\Program Files\\PowerShell\\7\\pwsh.exe -noprofile",
                "cursorColor": "#0ed6c6",
                "cursorShape": "emptyBox",
                "fontFace": "Cascadia Code",
                "fontSize": 14,
                "guid": "{c6783a98-47d9-4610-bc68-7a3f909e94fb}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{574e775e-4f2a-5b96-ac1e-a2962a402336}.png",
                "name": "PowerShell 7 No Profile",
                "padding": "0, 0, 0, 0",
                "snapOnInput": true,
                "startingDirectory": "C:\\",
                "tabTitle": "PowerShell 7 NoProfile",
                "backgroundImageAlignment": "bottomRight",
                "backgroundImageStretchMode": "none",
                "useAcrylic": true
            },
            {
                // Windows PowerShell - ok
                "acrylicOpacity": 0.5,
                "background": "#012456",
                "closeOnExit": true,
                "colorScheme": "Campbell",
                "commandline": "powershell.exe -nologo",
                "cursorColor": "#FFFFFF",
                "cursorShape": "underscore",
                "fontFace": "Consolas",
                "fontSize": 14,
                "guid": "{2bddf50b-a2af-451e-afa4-e1c3251fe3c5}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{61c54bbd-c2c6-5271-96e7-009a87ff44bf}.png",
                "name": "Windows PowerShell",
                "padding": "0, 0, 0, 0",
                "snapOnInput": true,
                "startingDirectory": "%USERPROFILE%",
                "tabTitle": "Windows PowerShell",
                "useAcrylic": false
            },
            {
                // Windows PowerShell No Profile -ok
                "acrylicOpacity": 0.5,
                "background": "#03396C",
                "backgroundImage": "C:\\Users\\Jeff\\Pictures\\powershell-2.jpg",
                "backgroundImageOpacity": 0.5,
                "backgroundImageStretchMode": "uniform",
                "backgroundImageAlignment": "right",
                "closeOnExit": true,
                "colorScheme": "Jeff",
                "commandline": "powershell.exe -nologo -noprofile",
                "cursorColor": "#03396C",
                "cursorHeight": 25,
                "cursorShape": "vintage",
                "fontFace": "Consolas",
                "fontSize": 16,
                "guid": "{ca3da2bd-de18-40f7-9fc6-234345d1e89f}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{61c54bbd-c2c6-5271-96e7-009a87ff44bf}.png",
                "name": "Windows PowerShell No Profile",
                "padding": "10, 10, 5, 10",
                "snapOnInput": true,
                "startingDirectory": "C:\\",
                "tabTitle": "PS No Profile",
                "useAcrylic": true
            },
            {
                // DOM1 - ok
                "acrylicOpacity": 0.5,
                "background": "#0a52e2",
                "closeOnExit": true,
                "colorScheme": "Jeff",
                "commandline": "powershell.exe -nologo -noprofile -noexit -file c:\\scripts\\wtpsremote.ps1 -computername dom1 -remoteprofile c:\\scripts\\remoteprofile.ps1",
                "cursorColor": "#FFFFFF",
                "cursorShape": "underscore",
                "fontFace": "Cascadia Code",
                "fontSize": 14,
                "guid": "{b89c512a-cd2b-4fb4-a6d2-3366cee85970}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{61c54bbd-c2c6-5271-96e7-009a87ff44bf}.png",
                "name": "DOM1",
                "padding": "0, 0, 0, 0",
                "snapOnInput": true,
                "startingDirectory": "%USERPROFILE%",
                "tabTitle": "PSRemote: DOM1.Company.Pri",
                "useAcrylic": false
            },
            {
                // Fred - ok
                "acrylicOpacity": 0.8,
                "closeOnExit": true,
                "colorScheme": "Campbell Powershell",
                "commandline": "C:\\Program Files\\PowerShell\\7\\pwsh.exe -nologo -noprofile -noexit -file c:\\scripts\\wtpsremote.ps1 -hostname fred-company-com -username jeff -remoteprofile c:\\scripts\\lxprofile.ps1",
                "cursorColor": "#FFFFFF",
                "cursorShape": "underscore",
                "fontFace": "Cascadia Code",
                "fontSize": 14,
                "guid": "{5668446e-620a-448f-874b-db766831b2ab}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{574e775e-4f2a-5b96-ac1e-a2962a402336}.png",
                "name": "Fred-ora",
                "padding": "0, 0, 0, 0",
                "snapOnInput": true,
                "startingDirectory": "C:\\",
                "tabTitle": "PSRemote: Fred",
                "backgroundImage": "C:\\users\\jeff\\Pictures\\smoking-penguin.jpg",
                "backgroundImageAlignment": "center",
                "backgroundImageStretchMode": "none",
                "backgroundImageOpacity": 0.3,
                "useAcrylic": true
            },
            {
                // CMD.exe - OK
                "acrylicOpacity": 1.0,
                "backgroundImage": "",
                "backgroundImageOpacity": 0.5,
                "backgroundImageStretchMode": "uniform",
                "closeOnExit": true,
                "colorScheme": "crt",
                "commandline": "cmd.exe",
                "cursorColor": "#0cf552",
                "cursorShape": "vintage",
                "experimental.retroTerminalEffect": true,
                "fontFace": "Cascadia Code",
                "fontSize": 20,
                "guid": "{e4f9024a-6bc8-4b11-be92-aa935bd79b35}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{0caa0dad-35be-5f56-a8ff-afceeeaa6101}.png",
                "name": "CMD",
                "padding": "0, 0, 0, 0",
                "snapOnInput": true,
                "startingDirectory": "%USERPROFILE%",
                "useAcrylic": true
            },
            {
                //Ubuntu 18.04 -ok
                "acrylicOpacity": 0.5,
                "closeOnExit": true,
                "backgroundImage": "C:\\Users\\Jeff\\Pictures\\computer-row.png",
                "backgroundImageOpacity": 0.3,
                "backgroundImageStretchMode": "fill",
                "colorScheme": "Jeff",
                "commandline": "wsl.exe -d Ubuntu-18.04",
                "cursorColor": "#FFFFFF",
                "cursorShape": "vintage",
                "fontFace": "Cascadia Code",
                "fontSize": 16,
                "guid": "{e64236d3-807c-49cb-90d8-469d392951f8}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{9acb9455-ca41-5af7-950f-6bca1bc9722f}.png",
                "tabTitle": "Ubuntu 18.04",
                "name": "Ubuntu-18.04",
                "padding": "0, 0, 0, 0",
                "snapOnInput": true,
                "useAcrylic": false
            },
            {
                //Netsh - ok
                "acrylicOpacity": 0.80000001192092896,
                "closeOnExit": true,
                "colorScheme": "Tango Dark",
                "commandline": "cmd /c netsh",
                "cursorColor": "#800006",
                "cursorHeight": 25,
                "cursorShape": "vintage",
                "fontFace": "Consolas",
                "fontSize": 18,
                "guid": "{24fe3b4f-d7de-444b-9b71-b946818c98a7}",
                "historySize": 9001,
                "icon": "ms-appx:///ProfileIcons/{0caa0dad-35be-5f56-a8ff-afceeeaa6101}.png",
                "name": "Netsh",
                "padding": "10, 0, 0, 0",
                "snapOnInput": true,
                "useAcrylic": true,
                "hidden": false
            },
            {
                "guid": "{a3a2e83a-884a-5379-baa8-16f193a13b21}",
                "hidden": false,
                "name": "PowerShell 7 Preview",
                "source": "Windows.Terminal.PowershellCore"
            },
            {
                "guid": "{b453ae62-4e3d-5e58-b989-0a998ec441b8}",
                "hidden": true,
                "useAcrylic": false,
                "name": "Azure Cloud Shell",
                "source": "Windows.Terminal.Azure"
            },
            {
                "guid": "{574e775e-4f2a-5b96-ac1e-a2962a402336}",
                "hidden": true,
                "name": "PowerShell",
                "source": "Windows.Terminal.PowershellCore"
            },
            {
                "guid": "{6e9fa4d2-a4aa-562d-b1fa-0789dc1f83d7}",
                "hidden": true,
                "name": "Legacy",
                "source": "Windows.Terminal.Wsl"
            },
            {
                "guid": "{c6eaf9f4-32a7-5fdc-b5cf-066e8a4b1e40}",
                "hidden": true,
                "name": "Ubuntu-18.04",
                "source": "Windows.Terminal.Wsl"
            }
        ]
    },

    // Add custom color schemes to this array.
    // To learn more about color schemes, visit https://aka.ms/terminal-color-schemes
    "schemes": [
        {
            "background": "#0C0C0C",
            "black": "#0C0C0C",
            "blue": "#0037DA",
            "brightBlack": "#767676",
            "brightBlue": "#3B78FF",
            "brightCyan": "#61D6D6",
            "brightGreen": "#16C60C",
            "brightPurple": "#B4009E",
            "brightRed": "#E74856",
            "brightWhite": "#F2F2F2",
            "brightYellow": "#F9F1A5",
            "cyan": "#3A96DD",
            "foreground": "#CCCCCC",
            "green": "#13A10E",
            "name": "Campbell",
            "purple": "#881798",
            "red": "#C50F1F",
            "white": "#CCCCCC",
            "yellow": "#C19C00"
        },
        {
            "name": "Campbell Powershell",
            "foreground": "#CCCCCC",
            "background": "#012456",
            "cursorColor": "#FFFFFF",
            "black": "#0C0C0C",
            "red": "#C50F1F",
            "green": "#13A10E",
            "yellow": "#C19C00",
            "blue": "#0037DA",
            "purple": "#881798",
            "cyan": "#3A96DD",
            "white": "#CCCCCC",
            "brightBlack": "#767676",
            "brightRed": "#E74856",
            "brightGreen": "#16C60C",
            "brightYellow": "#F9F1A5",
            "brightBlue": "#3B78FF",
            "brightPurple": "#B4009E",
            "brightCyan": "#61D6D6",
            "brightWhite": "#F2F2F2"
        },
        {
            "background": "#FFA500",
            "black": "#0C0C0C",
            "blue": "#0037DA",
            "brightBlack": "#767676",
            "brightBlue": "#3B78FF",
            "brightCyan": "#61D6D6",
            "brightGreen": "#16C60C",
            "brightPurple": "#B4009E",
            "brightRed": "#E74856",
            "brightWhite": "#F2F2F2",
            "brightYellow": "#F9F1A5",
            "cyan": "#3A96DD",
            "foreground": "#000000",
            "green": "#13A10E",
            "name": "Halloween",
            "purple": "#881798",
            "red": "#C50F1F",
            "white": "#CCCCCC",
            "yellow": "#C19C00"
        },
        {
            "background": "#0c0800",
            "black": "#0C0C0C",
            "blue": "#0037DA",
            "brightBlack": "#767676",
            "brightBlue": "#3B78FF",
            "brightCyan": "#61D6D6",
            "brightGreen": "#16C60C",
            "brightPurple": "#B4009E",
            "brightRed": "#E74856",
            "brightWhite": "#F2F2F2",
            "brightYellow": "#F9F1A5",
            "cyan": "#3A96DD",
            "foreground": "#0cf552",
            "green": "#13A10E",
            "name": "crt",
            "purple": "#881798",
            "red": "#C50F1F",
            "white": "#CCCCCC",
            "yellow": "#C19C00"
        },
        {
            "background": "#0C0C0C",
            "black": "#0C0C0C",
            "blue": "#0037DA",
            "brightBlack": "#767676",
            "brightBlue": "#3B78FF",
            "brightCyan": "#61D6D6",
            "brightGreen": "#16C60C",
            "brightPurple": "#B4009E",
            "brightRed": "#E74856",
            "brightWhite": "#F2F2F2",
            "brightYellow": "#F9F1A5",
            "cyan": "#3A96DD",
            "foreground": "#F2F5A9",
            "green": "#13A10E",
            "name": "Jeff",
            "purple": "#881798",
            "red": "#C50F1F",
            "white": "#CCCCCC",
            "yellow": "#C19C00"
        },
        {
            "background": "#282C34",
            "black": "#282C34",
            "blue": "#61AFEF",
            "brightBlack": "#5A6374",
            "brightBlue": "#61AFEF",
            "brightCyan": "#56B6C2",
            "brightGreen": "#98C379",
            "brightPurple": "#C678DD",
            "brightRed": "#E06C75",
            "brightWhite": "#DCDFE4",
            "brightYellow": "#E5C07B",
            "cyan": "#56B6C2",
            "foreground": "#DCDFE4",
            "green": "#98C379",
            "name": "One Half Dark",
            "purple": "#C678DD",
            "red": "#E06C75",
            "white": "#DCDFE4",
            "yellow": "#E5C07B"
        },
        {
            "background": "#FAFAFA",
            "black": "#383A42",
            "blue": "#0184BC",
            "brightBlack": "#4F525D",
            "brightBlue": "#61AFEF",
            "brightCyan": "#56B5C1",
            "brightGreen": "#98C379",
            "brightPurple": "#C577DD",
            "brightRed": "#DF6C75",
            "brightWhite": "#FFFFFF",
            "brightYellow": "#E4C07A",
            "cyan": "#0997B3",
            "foreground": "#383A42",
            "green": "#50A14F",
            "name": "One Half Light",
            "purple": "#A626A4",
            "red": "#E45649",
            "white": "#FAFAFA",
            "yellow": "#C18301"
        },
        {
            "background": "#002B36",
            "black": "#073642",
            "blue": "#268BD2",
            "brightBlack": "#002B36",
            "brightBlue": "#839496",
            "brightCyan": "#93A1A1",
            "brightGreen": "#586E75",
            "brightPurple": "#6C71C4",
            "brightRed": "#CB4B16",
            "brightWhite": "#FDF6E3",
            "brightYellow": "#657B83",
            "cyan": "#2AA198",
            "foreground": "#839496",
            "green": "#859900",
            "name": "Solarized Dark",
            "purple": "#D33682",
            "red": "#DC322F",
            "white": "#EEE8D5",
            "yellow": "#B58900"
        },
        {
            "background": "#FDF6E3",
            "black": "#073642",
            "blue": "#268BD2",
            "brightBlack": "#002B36",
            "brightBlue": "#839496",
            "brightCyan": "#93A1A1",
            "brightGreen": "#586E75",
            "brightPurple": "#6C71C4",
            "brightRed": "#CB4B16",
            "brightWhite": "#FDF6E3",
            "brightYellow": "#657B83",
            "cyan": "#2AA198",
            "foreground": "#657B83",
            "green": "#859900",
            "name": "Solarized Light",
            "purple": "#D33682",
            "red": "#DC322F",
            "white": "#EEE8D5",
            "yellow": "#B58900"
        }
    ],

    // Add custom keybindings to this array.
    // To unbind a key combination from your defaults.json, set the command to "unbound".
    // To learn more about keybindings, visit https://aka.ms/terminal-keybindings
    "keybindings":
    [
        // Copy and paste are bound to Ctrl+Shift+C and Ctrl+Shift+V in your defaults.json.
        // These two lines additionally bind them to Ctrl+C and Ctrl+V.
        // To learn more about selection, visit https://aka.ms/terminal-selection
        { "command": {"action": "copy", "singleLine": false }, "keys": "ctrl+c" },
        { "command": "paste", "keys": "ctrl+v" },

        // Press Ctrl+Shift+F to open the search box
        { "command": "find", "keys": "ctrl+shift+f" },

        // Press Alt+Shift+D to open a new pane.
        // - "split": "auto" makes this pane open in the direction that provides the most surface area.
        // - "splitMode": "duplicate" makes the new pane use the focused pane's profile.
        // To learn more about panes, visit https://aka.ms/terminal-panes
        { "command": { "action": "splitPane", "split": "auto", "splitMode": "duplicate" }, "keys": "alt+shift+d" },
        {
            // Ctrl+Shift+U will launch the Ubuntu profile
            "command": {
                "action": "newTab",
                "profile": "Ubuntu-18.04"
            },
            "keys": [
                "ctrl+shift+u"
            ]
        },
        {
            "command": {
                "action": "splitPane",
                "split": "auto",
                "profile": "PowerShell 7 No Profile",
                "startingDirectory": "C:\\",
                "tabTitle": "PS7 Clean"
            },
            "keys": [
                "ctrl+shift+>"
            ]
        },
        {
            "command": {
                "action": "splitPane",
                "split": "auto",
                "profile": "Windows PowerShell No Profile",
                "startingDirectory": "C:\\",
                "tabTitle": "PS Clean"
            },
            "keys": [
                "ctrl+shift+<"
            ]
        }
    ]
}
```

**Comment by jdhitsolutions** [comment]
One thing I am doing is that I have a symbolic link that points to settings.json in my OneDrive folder. But I was doing that before and everything worked as expected.

My profile settings has been moved along from very early builds. So it is possible I keep moving over crud.

**Comment by DHowett-MSFT** [comment]
Ah. Symbolic links. Terminal's never been able to hot reload settings behind symbolic links, because to do so would require it to open a handle to the _destination directory_ (the whole directory, which could produce an immense number of file change notifications if (for example) it's your user folder or some other high-traffic area.)

**Comment by DHowett-MSFT** [comment]
(If it worked before, it was almost certainly a fluke.)

**Comment by jdhitsolutions** [comment]
And somehow, I'm getting additional profiles that shouldn't be enabled. In my settings they are set as hidden but I'm still seeing them. I tried deleting them but they came back on restart.  Is defaults.json getting merged with my settings.json?

**Comment by jdhitsolutions** [comment]
The reason for the link is primarily to make sure I have a backup and to make it easy to share my file across the different systems I use.

**Comment by DHowett-MSFT** [comment]
Yeah -- defaults.json is always loaded and your settings are applied on to.

However, you are in a very interesting position. It looks like you're one of the very few people who have _pre-0.2_ profiles. Before 0.2 shipped, we stabilized the GUIDs for the powershell and cmd profiles.

In 0.5, we used those stable guids to ensure that use customizations to those profiles _overwrote_ the ones in `defaults.json` (without creating duplicates.)

When 0.5 rolled around, people who had the stable GUIDs saw no change. People who had pre-0.2 settings would see copies of cmd and Windows PowerShell. Since nobody was using versions < 0.2 (we hadn't yet released the prebuilt binaries), we deemed this acceptable.

If you change the GUIDs for Windows PowerShell and cmd to `{61c54bbd-c2c6-5271-96e7-009a87ff44bf}` and `{0caa0dad-35be-5f56-a8ff-afceeeaa6101}` respectively, your versions of those profiles will override instead of adding to the existing ones.

If you merge the "powershell 7 preview" and "PowerShell 7 - ok" profiles by giving the "ok" one the `guid` and `source` from the lower-down one, that profile will override instead of adding another copy of the powershell profile.

**Comment by DHowett-MSFT** [comment]
If you really want to suppress all the dynamic profiles (sorry), you can add a `disabledProfileSources` key to the top level of settings -- the schema should guide you if you're using a schema-enabled editor. It's an array that tells us which `source`s to straight-up ignore and stop putting auto-profiles in for.

**Comment by jdhitsolutions** [comment]
I noticed the GUID changes but must have missed the implications.  I guess I'll need to spend a bit more time cleaning up my profile. That's what I get I suppose for being an enthusiastic early adopter. :wink:

**Comment by jdhitsolutions** [comment]
I went back to a file without the links and now this works as expected. You can close this if you no longer need the issue for anything.,

**Comment by DHowett-MSFT** [comment]
Thanks for sticking with us for so long!

If you want to take the opportunity, you can also move a bunch of those repetitive settings into the `defaults` block -- we wanted to streamline settings so you didn't need to have so many copies of everything and so many places to update when you, like, got a new favorite font or something. :smile:

**Comment by jdhitsolutions** [comment]
I am testing with a hardlink and that appears to be working.

**Comment by jdhitsolutions** [comment]
I use the heck out of profiles. I need PowerShell 7 and Windows PowerShell, both normal and also sessions with no profile scripts. Then there's PowerShell preview builds. Plus some custom ones I came up with to open PowerShell remoting sessions to other systems and VMs. I'm probably pushing the profile envelope.

**Comment by Firehawke** [comment]
> 
> 
> Yeah -- defaults.json is always loaded and your settings are applied on top.
> 
... 
> In 0.5, we used those stable guids to ensure that use customizations to those profiles _overwrote_ the ones in `defaults.json` (without creating duplicates.)
> 
> When 0.5 rolled around, people who had the stable GUIDs saw no change. People who had pre-0.2 settings would see copies of cmd and Windows PowerShell. Since nobody was using versions < 0.2 (we hadn't yet released the prebuilt binaries), we deemed this acceptable.
> 
> If you change the GUIDs for Windows PowerShell and cmd to `{61c54bbd-c2c6-5271-96e7-009a87ff44bf}` and `{0caa0dad-35be-5f56-a8ff-afceeeaa6101}` respectively, your versions of those profiles will override instead of adding to the existing ones.

This just kind of bit me. I'd previously overwritten Windows PowerShell with PowerShell 7 and the update had unexpected results. I was about to open an issue to ask about why defaults was overriding me, but your description here explains precisely why it was doing what it did, and most importantly what I needed to do to fix things--

I'll add that here in case anyone else runs into this issue:
 
I put PowerShell 7 back onto the GUID that WT assigned it (pulling it from the entry it put at the bottom of the sessions list) and left it at the top of the list. I then took the Windows PowerShell entry that it recreated, set it to hidden, and left it at the bottom of the list along with the Azure Cloud Shell entry I also wasn't needing.

Oh, and I changed the default session entry to the new GUID for PS7.

These changes caused WPS to no longer appear in the menus, no additional sessions created every time I saved the config, and PS7 stayed as the default at the top.

The interactions between default.json and settings.json aren't entirely obvious, so it probably wouldn't hurt to document that GUIDs are now static for certain standard entries and that those will be recreated if removed, so it's better to just override them with a "hidden": true entry.
